### DISCUSSION TITLE
Should UdpSocket::bind() be async?

### DISCUSSION THREAD

**Discussion by incisakura** [tokio-rs/tokio#4517] [General] [answered]
The only reason that these two functions are async is to resolve the `addr` as a domain name.
But is there anyone who binds a socket with the domain name?

Should it be better to make these functions to be non-async functions?

And should we add more `#[inline]` tags here?

https://github.com/tokio-rs/tokio/blob/43c224ff47e41628ca787d116080d69bd7030c3f/tokio/src/net/udp.rs#L149-L166

**Comment by Darksonn** [comment, marked as answer]
We could argue about this, but there's not much point. It can't be changed now regardless.

**Reply by incisakura** [reply]
Thanks for your answer. I wish this would be "fix" oneday.
